src/AppInstallerCLICore/Workflows/PortableFlow.cpp

```diff
@@ -196,7 +196,7 @@ namespace AppInstaller::CLI::Workflow
             for (const auto& nestedInstallerFile : nestedInstallerFiles)
             {
                 const std::filesystem::path& targetPath = targetInstallDirectory / ConvertToUTF16(nestedInstallerFile.RelativeFilePath);
-                
+
```

Comment:
For the multi portable inside zip case, I think we can do the following: (basically merging the 2 for loops)
```
for (const auto& entry : std::filesystem::directory_iterator(installerPath))
{
   if (std::filesystem::is_directory(entryPath))
   {
        entries.emplace_back(std::move(PortableFileEntry::CreateDirectoryEntry(entryPath, targetPath)));
   }
   else
   {
        if (file is found in NestedInstallerFiles && !nestedInstallerFile.PortableCommandAlias.empty())
        {
             // Update targetPath to use the PortableCommandAlias as file name
             Filesystem::AppendExtension(commandAlias, ".exe");
             entries.emplace_back(std::move(PortableFileEntry::CreateSymlinkEntry(symlinkDirectory / commandAlias, targetPath)));
        }
        entries.emplace_back(std::move(PortableFileEntry::CreateFileEntry(entryPath, targetPath, {})));
   }
}


```

Reply:
I'm wondering how that would impact performance, if every file in a directory is checked against the portable file entries

Reply:
yeah, but this is the best I can think of to easily make the file and symlink in sync. I was thinking it's just relative path string comparison (not checking the actual filesystem path) so hopefully the performance is not impacted a lot.

Reply:
> yeah, but this is the best I can think of to easily make the file and symlink in sync. I was thinking it's just relative path string comparison (not checking the actual filesystem path) so hopefully the performance is not impacted a lot.

So I did try this, and I think that when the `entryPath` is a directory, it never recurses into the directory to get the files, it simply copies the entire directory over at once. Here's what I got for code -
```
                if (std::filesystem::is_directory(entryPath))
                {
                    entries.emplace_back(std::move(PortableFileEntry::CreateDirectoryEntry(entryPath, targetPath)));
                }
                else
                {
                    for (const auto& nestedInstallerFile : nestedInstallerFiles)
                    {
                        std::filesystem::path relativeNestedPath = Utility::ConvertToUTF16(nestedInstallerFile.RelativeFilePath);
                        relativeNestedPath = relativeNestedPath.lexically_normal();
                        if (relativeNestedPath.compare(relativePath.lexically_normal()) == 0)
                        {
                            if (!nestedInstallerFile.PortableCommandAlias.empty())
                            { 
                                targetPath.replace_filename(Utility::ConvertToUTF16(nestedInstallerFile.PortableCommandAlias));
                                Filesystem::AppendExtension(targetPath, ".exe");
                            }
                            entries.emplace_back(std::move(PortableFileEntry::CreateSymlinkEntry(symlinkDirectory / targetPath.stem(), targetPath)));
                        }
                    }
                    entries.emplace_back(std::move(PortableFileEntry::CreateFileEntry(entryPath, targetPath, {})));
                }
```
(Pushed it to [renamePortable_Multiple](https://github.com/Trenly/winget-cli/blob/0bf642013adc0bc0cf3f3a0b82af9e2f84404481/src/AppInstallerCLICore/Workflows/PortableFlow.cpp#L190-L213) in case you want to test yourself)

Using a manifest like `Gyan.FFmpeg` (modified command aliases locally to test), no aliases are ever added and the files aren't renamed.
```
PS D:\Git\winget-cli> wingetdev install -m 'D:\Git\winget-pkgs\manifests\g\Gyan\FFmpeg\6.0' --force
This application is licensed to you by its owner.
Successfully verified installer hash
Extracting archive...
Successfully extracted archive
Starting package install...
Successfully installed
PS D:\Git\winget-cli> wing
```

But when the files are at the root level like with `CloudFoundry.CLI.v8`, then it does add the aliases and the files are renamed
```
PS D:\Git\winget-cli> wingetdev install -m 'D:\Git\winget-pkgs\manifests\c\CloudFoundry\CLI\v8\8.6.1\'
Found Cloud Foundry CLI [CloudFoundry.CLI.v8] Version 8.6.1
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Extracting archive...
Starting package install...
Command line alias added: "cfa"
Command line alias added: "cf8a"
Successfully installed
```

Reply:
Ah, I see. Thanks for trying out. Then I guess the CreateDirectoryEntry() struct needs to contain a member for a list of rename files and update PortableInstaller::InstallFile() to act accordingly. The entries will have to be sorted to process symlink entries last. Which is unfortunately a bit more work.

Reply:
I got lost about halfway through that, but I'll believe you. 

I'm assuming that a separate PR would be best for that work?

Reply:
If you are not working on it now, I think it's ok to get this in first. Please let me know if you wish me to continue for the multi file scenario as I want the whole feature working during winget 1.6.

Reply:
Alternatively, would a "Rename" entry work?  Since the entries would need to be sorted anyways, a new entry type could be added to rename the aliased files after the Directory and File entries are processed, but before the Symlinks are processed?

Reply:
> If you are not working on it now, I think it's ok to get this in first. Please let me know if you wish me to continue for the multi file scenario as I want the whole feature working during winget 1.6.

I should have time to work on it tomorrow

Reply:
> Alternatively, would a "Rename" entry work? Since the entries would need to be sorted anyways, a new entry type could be added to rename the aliased files after the Directory and File entries are processed, but before the Symlinks are processed?

Yes, it could work. But I feel there'll be multiple modifications to portable index code to handle the new rename entry (basically ignore it). And for existing file entry, with new rename entry, it'll need to handle recording correct value to the index.

So I feel adding a vector of {relativePath, renamedRelativePath} to struct PortableFileEntry (this vector can only be used by Type::Directory), and update the PortableInstaller::InstallFile() will be cleanest. This requires zero portable index change.

Reply:
I see; Seems you know the dependencies on the portable index much better than I do; I think it's probably best if I don’t try to make that deep of code change myself